modules/jwplayerBidAdapter.js

```diff
@@ -50,4 +87,129 @@ export const spec = {
+  // Open RTB Request Object
+  const openrtbRequest = {
+    id: bidRequest.params.bidId,
+    imp: buildRequestImpression(bidRequest, bidderRequest),
+    site: buildRequestSite(bidRequest),
+    device: buildRequestDevice()
+  };
+
+  // Attaching GDPR Consent Params
+  if (bidderRequest.gdprConsent) {
+    deepSetValue(openrtbRequest, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    deepSetValue(openrtbRequest, 'regs.ext.gdpr', (bidderRequest.gdprConsent.gdprApplies ? 1 : 0));
+  }
+
+  // CCPA
+  if (bidderRequest.uspConsent) {
+    deepSetValue(openrtbRequest, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+  }
+
+  return JSON.stringify(openrtbRequest);;
+}
+
+function buildRequestImpression(bidRequest) {
+  const impressions = [];
+
+  const impressionObject = {
+    id: bidRequest.adUnitCode,
+    secure: isSecure() ? 1 : 0
+  };
+
+  impressionObject.video = buildImpressionVideo(bidRequest);
+
+  const bidFloorData = buildBidFloorData(bidRequest);
+  impressionObject.bidfloor = bidFloorData.floor;
+  impressionObject.bidfloorcur = bidFloorData.currency;
+  
+  impressionObject.ext = buildImpressionExtension(bidRequest);
+
+  impressions.push(impressionObject);
+
+  return impressions;
+}
+
+function buildImpressionVideo(bidRequest) {
+  const videoParams = deepAccess(bidRequest, 'mediaTypes.video', {});
+
+  const video = {};
+
+  // Obtain all ORTB params related video from Ad Unit
+  VIDEO_ORTB_PARAMS.forEach((param) => {
+    if (videoParams.hasOwnProperty(param)) {
+      video[param] = videoParams[param];
+    }
+  });
+
+  return video;
+}
+
+function buildImpressionExtension(bidRequest) {
+  return {
+    appnexus: {
+      placement_id: bidRequest.params.placementId
+    }
+  };
+}
+
+function buildBidFloorData(bidRequest) {
+  const {params} = bidRequest;
+  // Bid Floor
+  const bidFloorRequest = {
+    currency: params.currency || 'USD',
+    mediaType: 'video',
+    size: '*'
+  };
+
+  let floorData;
+  if (isFn(bidRequest.getFloor)) {
+    floorData = bidRequest.getFloor(bidFloorRequest);
+  } else if (params.bidfloor) {
+    floorData = {floor: params.bidfloor, currency: params.currency || 'USD'};
+  }
+
+  return floorData;
+}
+
+function buildRequestSite(bidRequest) {
+  const site = {
+    domain: window.location.hostname,
+    page: window.location.href,
+    ref: bidRequest.refererInfo ? bidRequest.refererInfo.referer || null : null
+  };
+
+  const videoParams = deepAccess(bidRequest, 'mediaTypes.video', {});
+
+  // Site Content
+  if (videoParams.content && isPlainObject(videoParams.content)) {
+    openrtbRequest.site.content = {};
+    const contentStringKeys = ['id', 'title', 'series', 'season', 'genre', 'contentrating', 'language', 'url'];
+    const contentNumberkeys = ['episode', 'prodq', 'context', 'livestream', 'len'];
+    const contentArrayKeys = ['cat'];
+    const contentObjectKeys = ['ext'];
+    for (const contentKey in videoBidderParams.content) {
```

Comment:
i don't think `videoBidderParams` is ever defined

Reply:
fixed.